packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -67,6 +89,31 @@ Each column that is sorted will have an appropriate icon displayed in the header
 
 Each column that is sorted ascending will have `aria-sort="ascending"` set on the header, and each column that is sorted descending will have `aria-sort="descending"` set on the header.
 
+### Interactive Sorting UX
+
+The UX/ interactions for interactive sorting are captured in [#885: UX Design for basic column/row interactions](https://github.com/ni/nimble/issues/885).  
+To summarize:
+
+If the table `sortMode` is `single` or `multiple`:  
+Single-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were also sorted become unsorted
+-   The clicked column gets a sort index of `0` when sorted, and `undefined` if it's transitioning back to unsorted.
+
+If the table `sortMode` is `multiple` only:  
+Shift-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were already sorted maintain their sorting
+-   If the clicked column was already sorted, its sort index does not change
+-   If the clicked column is transitioning to unsorted, its sort index will be set to `undefined`
+-   If the clicked column is transitioning to being sorted, it will get a sort index of `0` if no other columns were already sorted, or `max(alreadySortedColumn1SortIndex, ..., alreadySortedColumnNSortIndex) + 1` if there were existing sorted columns
+
+If interactive sorting is enabled, sorting menu items also appear in the column header menu:  
+![Sorting via Column Header Menu](./spec-images/HeaderMenuSorting.png)  
+Sorting via the menu will always unsort any other columns that were already sorted (TBD: even when unsorting the current column?)
+
+For columns with `sorting-disabled` set to true, clicking/Shift-clicking the column header will not affect the sort state, and the sort menu options will not appear in the column header menu.
```

Comment:
Only interactive sorting is prevented with this property right? Not programmatic sorting.

Is there an implication that all columns are sortable? ie for grouping column authors can opt-in to grouping. Is that not true for sorting? It's possible that we had not made that pattern yet so that's why sorting isn't column author configurable? @mollykreis

Reply:
Correct, only interactive sorting. I might need to rename the property to make that clearer (though `interactive-sorting-disabled` is a bit verbose)

I believe that all columns are at least programmatically sortable currently, as sortDirection/sortIndex are on the base column type.

Reply:
Updated HLD to clarify here that programmatic sorting would still be possible.

Reply:
This sounds inconsistent with grouping. I believe that `grouping-disabled` turns off grouping for a column even when it is programmatically specified. Is there a reason we would want the grouping & sorting properties to behave differently? If so, I think we should name them differently somehow to make this distinction clear.

Reply:
Took a look at the grouping changes. As you said `grouping-disabled` affects programmatic grouping too so it probably makes sense to match that here.

Reply:
Updated HLD to state that `sorting-disabled` affects both interactive and programmatic sorting